faucet/valve.py

```diff
@@ -798,6 +800,31 @@ def ports_delete(self, port_nums, log_msg='down', keep_cache=False, other_valves
 
         return ofmsgs
 
+    def ports_flush(self):
+        """Used to flush ports and corresponding data.
+
+        Args: N/A
+        Returns: N/A
+        """
+        for port_num in self.dp.ports.keys():
+            if not self.dp.port_no_valid(port_num):
+                continue
+            port = self.dp.ports[port_num]
+            port.dyn_phys_up = False
+
+            self._set_port_status(port_num, False, time.time())
```

Comment:
Instead of generating time.time() here could you update _datapath_disconnect() in faucet.py to generate a `now` variable and pass this through to valve.datapath_disconnect().

This will make it similar to how _datapath_connect() works as well.

Reply:
That makes sense. Made necessary changes.